features/0351-generic-json-message/README.md

```diff
@@ -0,0 +1,114 @@
+# Aries RFC 0351: Generic JSON Message
+- Authors: [Filip Burlacu](filip.burlacu@securekey.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2019-12-16
+- Status Note: implementation is being explored by SecureKey
+- Supersedes:
+- Start Date: 2018-12-05
+- Tags: feature, protocol
+
+## Summary
+
+This protocol defines a format for sending arbitrary JSON payloads using DIDComm, for use cases where business logic or external (to agent) applications need to communicate over DIDComm.
+
+## Motivation
+
+Aries users may need to support communication that doesn't fit within a particular existing Aries protocol. In cases where machine-readable data needs to be transmitted, a defined format for sending such messages allows those users to transmit data over DIDComm, without making custom agents with custom DIDComm message formats that might collide with future, standard DIDComm message formats.
+
+Beyond allowing agents to implement custom protocols, this protocol allows clients that aren't Aries agents to communicate JSON messages over DIDComm using Aries agents as [mediators or relays](/concepts/0046-mediators-and-relays/README.md). 
```

Comment:
An additional motivation is to allow a generic message development, such that an Agent codebase doesn't need to be changed for each message type.

This concept works in situations where only the "business logic" (as opposed to the Agent) needs to maintain state. In other words, this concept works for messages where the Agent only needs to temporarily maintain threading state.